### PR TITLE
Add iterative searcher to Foldseek

### PR DESCRIPTION
**Pull request by martin-steinegger**

### CODE REVIEW

File: frontend/Search.vue [hunk 1]
```diff
@@ -168,14 +213,16 @@ export default {
         return {
             inSearch: false,
             errorMessage: "",
-            mode: storage.getItem('mode') || this.$STRINGS.MODE_DEFAULT_KEY,
+            showCurl: false,
+            mode: this.$STRINGS.MODE_DEFAULT_KEY,
```
Comment (milot-mirdita) [inline frontend/Search.vue:217]: Please restore the previous version

File: frontend/Search.vue [hunk 2]
```diff
@@ -35,102 +35,147 @@
...
+        <v-card-title primary-title class="pb-0 mb-0">
+            <div class="text-h5 mb-0">Reference</div>
+        </v-card-title>
+        <v-card-title primary-title class="pt-0 mt-0">
+            <p class="text-subtitle-2 mb-0" v-html="$STRINGS.CITATION"></p>
+        </v-card-title>
+        </v-card>
     </v-flex>
     </v-layout>
-    <reference :reference="$STRINGS.CITATION"></reference>
```
Comment (milot-mirdita) [inline frontend/Search.vue:132]: Please restore the previous version

File: frontend/Search.vue [hunk 3]
```diff
@@ -35,102 +35,147 @@
...
+                            <template v-if="database.length == 1">
+                                database
+                            </template>
+                            <template v-else>
+                                databases
+                            </template>
+                            ({{
+                                databases.filter(db => database.includes(db.path)).map(db => db.name).sort().join(", ")
+                            }})
+                        </template> with {{ $STRINGS.APP_NAME }} in <strong>{{ modes[mode] }}</strong> mode<span v-if="iterativeSearch"><strong> iterative</strong></span>.
```
Comment (milot-mirdita) [inline frontend/Search.vue:156]: Please add iterative to beginning of sentence

File: frontend/Search.vue [hunk 4]
```diff
@@ -35,102 +35,147 @@
...
+                    </div>
+                </panel>
+
+
+            </v-flex>
+            <v-flex>
+                <panel>
+                <template slot="content">
+                    <div class="actions" :style="!$vuetify.breakpoint.xsOnly ?'display:flex; align-items: center;' : null">
+                    <v-btn color="primary" :block="$vuetify.breakpoint.xsOnly" x-large v-on:click="search" :disabled="searchDisabled"><v-icon>{{ $MDI.Magnify }}</v-icon>&nbsp;Search</v-btn>
```
Comment (milot-mirdita) [inline frontend/Search.vue:136]: Please restore previous changes

File: frontend/Search.vue [hunk 5]
```diff
@@ -35,102 +35,147 @@
...
+                                <span>Improve sensitivity of search by performing an iterative search (--num-iterations 3).</span>
+                            </v-tooltip>
+
+                            <v-radio label="On" :value="true"></v-radio>
+                            <v-radio label="Off" :value="false"></v-radio>
+                        </v-radio-group>
+                    </div>
+                </panel>
+
+
```
Comment (milot-mirdita) [inline frontend/Search.vue:130]: Please remove empty newlines

File: frontend/Search.vue [hunk 6]
```diff
@@ -35,102 +35,147 @@
...
+                                        Iterative search
+                                        <v-icon color="#FFFFFFB3" style="margin-top:-3px" small v-on="on">{{ $MDI.HelpCircleOutline }}</v-icon>
+                                    </label>
+                                </template>
+                                <span>Improve sensitivity of search by performing an iterative search (--num-iterations 3).</span>
+                            </v-tooltip>
+
+                            <v-radio label="On" :value="true"></v-radio>
+                            <v-radio label="Off" :value="false"></v-radio>
+                        </v-radio-group>
```
Comment (milot-mirdita) [inline frontend/Search.vue:126]: Please keep email in addition to the new iterative search

File: frontend/Search.vue [hunk 7]
```diff
@@ -35,102 +35,147 @@
...
+                                    <label v-on="on">Mode&nbsp;<v-icon color="#FFFFFFB3" style="margin-top:-3px" small v-on="on">{{ $MDI.HelpCircleOutline }}</v-icon></label>
+                                </template>
+                                <span v-html="$STRINGS.MODE_HELP"></span>
+                            </v-tooltip>
+                            <v-radio hide-details
+                                    v-for="i in ($STRINGS.MODE_COUNT - 0)"
+                                    :key="i"
+                                    :value="$STRINGS['MODE_KEY_' + i]"
+                                    :label="$STRINGS['MODE_TITLE_' + i]">
+                            </v-radio>
```
Comment (milot-mirdita) [inline frontend/Search.vue:106]: Please revert the whitespace changes

File: frontend/Search.vue [hunk 8]
```diff
@@ -35,102 +35,147 @@
...
-                </v-tooltip>
-                <v-radio hide-details
-                            v-for="i in ($STRINGS.MODE_COUNT - 0)"
-                            :key="i"
-                            :value="$STRINGS['MODE_KEY_' + i]"
-                            :label="$STRINGS['MODE_TITLE_' + i]"
-                            ></v-radio>
-            </v-radio-group>
+                    <div slot="content">
+                        <!-- Mode Section -->
```
Comment (milot-mirdita) [inline frontend/Search.vue:93]: Please revert unrelated changes

File: frontend/Search.vue [hunk 9]
```diff
@@ -35,102 +35,147 @@
...
-                @update:all-databases="databases = $event"
-                :hideEmail="hideEmail"
-                ></databases>
+                        <div class="actions input-buttons-panel">
+                            <div class="input-buttons-left">
+                                <load-acession-button v-if="$APP == 'foldseek'" @select="query = $event" @loading="accessionLoading = $event" :preload-source="preloadSource" :preload-accession="preloadAccession"></load-acession-button>
+                                <file-button id="file" :label="$STRINGS.UPLOAD_LABEL" v-on:upload="upload"></file-button>
+                                <PredictStructureButton v-if="$APP == 'foldseek'" :query="query" v-model="predictable" v-on:predict="query = $event"></PredictStructureButton>
+                            </div>
+                            <file-button id="localFile" label="Upload previous results" @upload="uploadJSON"></file-button>
```
Comment (milot-mirdita) [inline frontend/Search.vue:44]: Please revert unrelated changes

File: frontend/Search.vue [hunk 10]
```diff
@@ -193,6 +240,18 @@ export default {
...
+        }
+        if (localStorageEnabled && localStorage.databases) {
+            this.databases = JSON.parse(localStorage.databases);
+        }
+        if (localStorageEnabled && localStorage.iterativeSearch) {
+            this.iterativeSearch = JSON.parse(localStorage.iterativeSearch);
+        }
+        if (localStorageEnabled && localStorage.taxFilter) {
+            this.taxFilter = JSON.parse(localStorage.taxFilter);
+        }
```
Comment (milot-mirdita) [inline frontend/Search.vue:254]: These should be set somewhere else, they were removed in the last refactoring

Comment (martin-steinegger) [reply frontend/Search.vue]: done

File: frontend/Search.vue (hunk 9 above)
Comment (martin-steinegger) [reply frontend/Search.vue]: done

File: frontend/Search.vue (hunk 8 above)
Comment (martin-steinegger) [reply frontend/Search.vue]: done

File: frontend/Search.vue (hunk 7 above)
Comment (martin-steinegger) [reply frontend/Search.vue]: done

File: frontend/Search.vue (hunk 6 above)
Comment (martin-steinegger) [reply frontend/Search.vue]: done

File: frontend/Search.vue (hunk 5 above)
Comment (martin-steinegger) [reply frontend/Search.vue]: done

File: frontend/Search.vue (hunk 4 above)
Comment (martin-steinegger) [reply frontend/Search.vue]: done

File: frontend/Search.vue (hunk 3 above)
Comment (martin-steinegger) [reply frontend/Search.vue]: why?

File: frontend/Search.vue (hunk 2 above)
Comment (martin-steinegger) [reply frontend/Search.vue]: done

File: frontend/Search.vue (hunk 1 above)
Comment (martin-steinegger) [reply frontend/Search.vue]: done
